cord19-explorer/src/main/java/edu/asu/diging/cord19/explorer/web/auth/CoordinateCleaningController.java

```diff
@@ -0,0 +1,134 @@
+                                    } else {
+                                        dd = Math.signum(d) * (Math.abs(d) + (m / 60.0) + (se / 3600.0));
+                                    }
+                                    formattedCoords.add(dd);
+                                    coords.clear();
+                                }
+                                if(coords.size() == 2) {
+                                    float d = Float.parseFloat(coords.get(0));
+                                    float m = Float.parseFloat(coords.get(1));
+                                    double dd = Math.signum(d) * (Math.abs(d) + (m / 60.0));
+                                    formattedCoords.add(dd);
+                                    coords.clear();
+                                }
+                                if(coords.size() == 1) {
+                                    Double doubleCoord = Double.parseDouble(coords.get(0));
+                                    formattedCoords.add(doubleCoord);
+                                    coords.clear();
+                                }
+                               
+                            } else if(s.equals("S") || s.equals("W")) {
+                                // handle negative coords
+                                if(coords.size() == 3) {
+                                    float d = Float.parseFloat(coords.get(0));
+                                    float m = Float.parseFloat(coords.get(1));
+                                    float se = Float.parseFloat(coords.get(2));
+                                    double dd = 0;
+                                    if(d == 0) {
+                                        dd = ((m / 60.0) + (se / 3600.0));
+                                    } else {
+                                        dd = Math.signum(d) * (Math.abs(d) + (m / 60.0) + (se / 3600.0));
+                                    }
+                                    dd = dd *-1;
+                                    formattedCoords.add(dd);
+                                    coords.clear();
+                                }
+                                if(coords.size() == 2) {
+                                    float d = Float.parseFloat(coords.get(0));
+                                    float m = Float.parseFloat(coords.get(1));
+                                    double dd = Math.signum(d) * (Math.abs(d) + (m / 60.0));
+                                    dd = dd *-1;
+                                    formattedCoords.add(dd);
+                                    coords.clear();
+                                }
+                                if(coords.size() == 1) {                
+                                    Double doubleCoord = Double.parseDouble(coords.get(0)) * -1;
+                                    formattedCoords.add(doubleCoord);
+                                    coords.clear();
```

Comment:
i'm fairly certain this can be combined with the above code to avoid duplicate code.